### ISSUE TITLE
Reconciler cannot recreate STS

### ISSUE THREAD

**Issue by oleksii-kalinin** [opensearch-project/opensearch-k8s-operator#663]
operator version: 2.4.0
os version: 2.8

{"level":"error","ts":"2023-11-20T10:08:20.314Z","msg":"Reconciler error","controller":"opensearchcluster","controllerGroup":"opensearch.opster.io","controllerKind":"OpenSearchCluster","OpenSearchCluster":{"name":"opensearch-cluster","namespace":"elastic-stack-logging"},"namespace":"elastic-stack-logging","name":"opensearch-cluster","reconcileID":"1fbc7f4e-836b-4b5e-b9d1-63f5f5ec78fb","error":"recreating object on immutable field change has to be enabled explicitly through the reconciler options: StatefulSet.apps \"opensearch-cluster-hot\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden; recreating object on immutable field change has to be enabled explicitly through the reconciler options: StatefulSet.apps \"opensearch-cluster-warm\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden","errorCauses":[{"error":"recreating object on immutable field change has to be enabled explicitly through the reconciler options: StatefulSet.apps \"opensearch-cluster-hot\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden","errorVerbose":"StatefulSet.apps \"opensearch-cluster-hot\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden\nrecreating object on immutable field change has to be enabled explicitly through the reconciler options\ngithub.com/cisco-open/operator-tools/pkg/reconciler.(*GenericResourceReconciler).ReconcileResource\n\t/go/pkg/mod/github.com/cisco-open/operator-tools@v0.30.0/pkg/reconciler/resource.go:515\nopensearch.opster.io/pkg/reconcilers.(*ClusterReconciler).reconcileNodeStatefulSet\n\t/workspace/pkg/reconcilers/cluster.go:379\nopensearch.opster.io/pkg/reconcilers.(*ClusterReconciler).Reconcile\n\t/workspace/pkg/reconcilers/cluster.go:121\nopensearch.opster.io/controllers.(*OpenSearchClusterReconciler).reconcilePhaseRunning\n\t/workspace/controllers/opensearchController.go:321\nopensearch.opster.io/controllers.(*OpenSearchClusterReconciler).Reconcile\n\t/workspace/controllers/opensearchController.go:142\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1594"},{"error":"recreating object on immutable field change has to be enabled explicitly through the reconciler options: StatefulSet.apps \"opensearch-cluster-warm\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden","errorVerbose":"StatefulSet.apps \"opensearch-cluster-warm\" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy', 'persistentVolumeClaimRetentionPolicy' and 'minReadySeconds' are forbidden\nrecreating object on immutable field change has to be enabled explicitly through the reconciler options\ngithub.com/cisco-open/operator-tools/pkg/reconciler.(*GenericResourceReconciler).ReconcileResource\n\t/go/pkg/mod/github.com/cisco-open/operator-tools@v0.30.0/pkg/reconciler/resource.go:515\nopensearch.opster.io/pkg/reconcilers.(*ClusterReconciler).reconcileNodeStatefulSet\n\t/workspace/pkg/reconcilers/cluster.go:379\nopensearch.opster.io/pkg/reconcilers.(*ClusterReconciler).Reconcile\n\t/workspace/pkg/reconcilers/cluster.go:121\nopensearch.opster.io/controllers.(*OpenSearchClusterReconciler).reconcilePhaseRunning\n\t/workspace/controllers/opensearchController.go:321\nopensearch.opster.io/controllers.(*OpenSearchClusterReconciler).Reconcile\n\t/workspace/controllers/opensearchController.go:142\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1594"}],"stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

**Comment by swoehrl-mw** [comment]
You need to provide context and details (the cluster spec you used, what did you do, what did you observe, ...). Just throwing log lines about will not get you a meaningful response.

**Comment by oleksii-kalinin** [comment]
If in logs

**Comment by oleksii-kalinin** [comment]
Context: `kubectl delete pods -l label=opensearch`
After deleting all pods (as well as when nodes are cycled) this error is thrown by the operator. 
No changes to the operator were made during delete - container creating. 
@swoehrl-mw

**Comment by oleksii-kalinin** [comment]
Addition:
- delete all nodes
- OS cluster is dead now
- securityconfig updater waits for the cluster (cluster dead in this time)
- the cluster won't come up

**Comment by prudhvigodithi** [comment]
[Triage]
Adding @@swoehrl-mw @Jochen Kressin @Piotr Chmielnik @Hendrik Saly @bbarani.

**Comment by piellick** [comment]
@oleksii-kalinin oleksii-kalinin we seem to have same problems with our cluster, have you find a solution ?

**Comment by oleksii-kalinin** [comment]
@piellick nope, but we updated to 2.5.1, and as for now, the OS cluster is pretty stable, and there are no errors in the manager.

**Comment by piellick** [comment]
@oleksii-kalinin OK thanks, we were allready in 2.5.1 and after delete all pod,  this error still appear.

**Comment by jackychanel** [comment]
> @oleksii-kalinin OK thanks, we were allready in 2.5.1 and after delete all pod, this error still appear.

Confirm that we have the same problem in 2.5.1, we will try to desactivate `manager.parallelRecoveryEnabled: false` ... let you know if this are going better after that

**Comment by albertobagnacani** [comment]
We have the same problem with version ´2.11.1´.

**Comment by piellick** [comment]
@albertobagnacani , @oleksii-kalinin did you find any solution ? We are still stuck on this problem.

**Comment by josedev-union** [comment]
@synhershko we can close this
